### PR TITLE
feat: add file overwrite feature

### PR DESCRIPTION
**Pull request by adityak74**
This adds support to overwrite the file if required. overwrite: true can be passed in to specify that the file with the same needs to be updated/overwritten. Google Drive will manage the versioning of the file being updated/uploaded.

### CODE REVIEW

**Comment by adityak74** [comment]
@mathisve - This PR should be good for a review!

File: LICENSE.md [hunk 1]
```diff
@@ -4,4 +4,4 @@ Permission is hereby granted, free of charge, to any person obtaining a copy of
 
 The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
 
-THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
\ No newline at end of file
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
```
Comment (crejaud) [inline LICENSE.md:7]: Your IDE is probably adding extra characters at the end of each file. You'd probably want to revert this change

File: main.go [hunk 2]
```diff
@@ -74,28 +115,37 @@ func main() {
...
+			githubactions.Fatalf(fmt.Sprintf("opening file with filename: %v failed with error: %v", filename, err))
+		}
 		name = file.Name()
 	}
 
-	f := &drive.File{
-		Name:    name,
-		Parents: []string{folderId},
+	if overwriteFlag {
+		r, err := svc.Files.List().Do()
```
Comment (crejaud) [inline main.go:127]: This is querying every single file in the directory, hence there is a bug in this diff. If there are files, but none of them match the `filename`, then nothing is uploaded to google drive. The expected behaviour would be to upload a new file.

Let's try querying the files by name instead using Q(). See https://pkg.go.dev/google.golang.org/api/drive/v3#FilesListCall.Q

Here is an explanation of the query syntax:
https://developers.google.com/drive/api/v3/search-files#query_string_examples

```
// Query for all files in google drive directory with name = <filename>
filenameQuery = fmt.Sprintf("name = '%s'", filename)
filesQueryCallResult, err := svc.Files.List().Q(filenameQuery).Do()

if len(filesQueryCallResult.Files == 0) {
  // upload file to drive
} else {
  // overwrite each file
}
```

File: README.md [hunk 3]
```diff
@@ -24,6 +28,11 @@ Required: **NO**
 
 The name you want the file to have in Google Drive. If this input is not provided, it will use the `filename`.
 
+## ``overwrite``
+Required: **NO**
+
+If you want to overwrite the filename with existing file, it will use the `filename`.
```
Comment (crejaud) [inline README.md:34]: Given that `filename` is the file that is stored locally, and `name` is the preferred name to be displayed on google drive, shouldn't this support overwriting based on `name`? And if `name` is not provided, then use `filename`.

Using the `overwrite` parameter now rends `name` useless. This doesn't seem to be the preferred behaviour.

File: main.go [hunk 4]
```diff
@@ -18,13 +23,40 @@ import (
...
-	scope = "https://www.googleapis.com/auth/drive.file"
-	filenameInput = "filename"
-	nameInput = "name"
-	folderIdInput = "folderId"
+	scope            = "https://www.googleapis.com/auth/drive.file"
+	filenameInput    = "filename"
+	nameInput        = "name"
+	folderIdInput    = "folderId"
 	credentialsInput = "credentials"
+	overwrite        = "false"
```
Comment (crejaud) [inline main.go:31]: These style changes seem out of scope for this diff

File: README.md [hunk 5]
```diff
@@ -62,8 +71,10 @@ jobs:
       - name: Upload to gdrive
         uses: team-tumbleweed/gdrive-upload-action@main
         with:
+          credentials: ${{ secrets.credentials }}
           filename: "archive.zip"
-          name: "documentation.zip"
           folderId: ${{ secrets.folderId }}
-          credentials: ${{ secrets.credentials }}
+          name: "documentation.zip" # optional string
```
Comment (crejaud) [inline README.md:77]: why the re-ordering?

**Review by crejaud — CHANGES_REQUESTED** [review]
Thanks for working on this. I have a use-case for an overwrite parameter as well here

**Comment by mathisve** [comment]
Closed due to #20.

Still want to thank you @adityak74, @Crejaud and @asfaltboy for contributing!

**Pull request closed without merging**